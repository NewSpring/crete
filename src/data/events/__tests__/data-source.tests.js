import ApollosConfig from '@apollosproject/config';
import { dataSource as EventDataSource } from '../index';

ApollosConfig.loadJs({
  ROCK_MAPPINGS: {
    PUBLIC_CALENDAR_ID: 1,
  },
});

describe('Event data sources', () => {
  let Event;
  beforeEach(() => {
    Event = new EventDataSource();
    Event.context = { dataSources: {} };
    Event.context.dataSources = {
      ContentItem: {
        getCoverImage: () => ({
          __typename: 'ImageMedia',
          key: 'coverImage',
          name: 'Cover Image',
          sources: [{ uri: 'https://fakeimages.com/?coolpic1' }],
        }),
      },
    };
  });
  it('gets an event image', async () => {
    Event.request = () => ({
      cache: () => ({
        filter: () => ({
          andFilter: () => ({
            first: () => ({
              id: 1,
              name: 'some event',
              attributeValues: { coverImage: { value: 'image1' } },
            }),
          }),
        }),
      }),
    });
    expect(await Event.getImage({ eventItemId: 1 })).toMatchSnapshot();
  });
  it('return null for the image with no attributes', async () => {
    Event.request = () => ({
      cache: () => ({
        filter: () => ({
          andFilter: () => ({
            first: () => ({
              id: 1,
              name: 'some event',
              attributeValues: {},
            }),
          }),
        }),
      }),
    });
    expect(await Event.getImage({ eventItemId: 1 })).toMatchSnapshot();
  });
});
