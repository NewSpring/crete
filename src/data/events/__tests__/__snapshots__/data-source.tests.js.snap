// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event data sources gets an event image 1`] = `
Object {
  "__typename": "ImageMedia",
  "key": "coverImage",
  "name": "Cover Image",
  "sources": Array [
    Object {
      "uri": "https://fakeimages.com/?coolpic1",
    },
  ],
}
`;

exports[`Event data sources return null for the image with no attributes 1`] = `null`;
